### PR TITLE
Integrate problem details into Solid client library

### PR DESCRIPTION
**Pull request by acoburn**
This integrates the `@inrupt/solid-client-errors` library into this codebase.

This keeps the `FetchError` class compatible with its current state but adds a new async function: `problemDetails()`, which allows a consumer of the library to access the underlying problem details data, if a server returns such data.

### CODE REVIEW

File: src/resource/resource.test.ts [hunk 1]
```diff
@@ -49,6 +50,23 @@ jest.spyOn(globalThis, "fetch").mockImplementation(
...
+      headers: response.headers,
+    });
+    this.facadeUrl = url;
+  }
+
+  get url(): string {
+    return this.facadeUrl;
+  }
+}
+
```
[the remark this answers is not in the document]
Comment (acoburn) [reply src/resource/resource.test.ts]: Resolved in https://github.com/inrupt/solid-client-js/pull/2458/commits/aa7cb4bec115cfe4190e22c8a6957b6e0f172098

File: src/resource/resource.ts [hunk 2]
```diff
@@ -273,8 +279,48 @@ export class FetchError extends SolidClientError {
     return this.response.statusText;
   }
 
+  async problemDetails(): Promise<ProblemDetails> {
```
Comment (jholleran) [inline src/resource/resource.ts:282]: This is different to the `problemDetails()` function in: https://github.com/inrupt/solid-client-errors-js/blob/main/src/http/httpError.ts#L83. This difference will mean clients will have to handle errors differently.

This should be `get problemDetails()` so a client can call it like this `error.problemDetails`. Also it should not be async.

**Review by jholleran — CHANGES_REQUESTED** [review]
There is one other change that the e2e tests are showing up around ACP tests that I'm working on and will add it here.

**Review by jholleran — APPROVED** [review]
@NSeydoux I've reverted that change I made as I now don't think its needed. I had another look and its currently not called anywhere internally where we would get an error response. In the two places it is called one is a HEAD request and the other does a status code check before calling the function.

**Review by NSeydoux — APPROVED** [review]

**Pull request merged**